src/modules/localization/src/frame_tf_calibrate/frame_calibration.cpp

```diff
@@ -193,18 +169,77 @@ namespace frame_calibration_node
                 }
                 else
                 {
-                    // calculate the euclidian distance between point (n) and point (n-1)
                     float distance_comparison;
                     float gps_threshold;
+                    float vehicle_heading_from_gps;
+                    float lpf_coeff;
+                    int sample_size;
 
+                    frameCalibrationNode_nh.getParam("/frame_calibration/lpf_coeff", lpf_coeff);
                     frameCalibrationNode_nh.getParam("/frame_calibration/gps_threshold", gps_threshold);
-                    distance_comparison = std::sqrt(std::pow(geodesy_tf_msg.pose.pose.position.x - previous_geodesy_tf_msg.pose.pose.position.x, 2) + std::pow(geodesy_tf_msg.pose.pose.position.y - previous_geodesy_tf_msg.pose.pose.position.y, 2));
+                    frameCalibrationNode_nh.getParam("/frame_calibration/sample_size", sample_size);
+                    // calculate the euclidian distance between point (n) and point (n-1)
+                    distance_comparison = std::sqrt(std::pow(geodesy_tf_msg.pose.pose.position.x - previous_geodesy_tf_msg.pose.pose.position.x, 2) + std::pow(geodesy_tf_msg.pose.pose.position.y - previous_geodesy_tf_msg.pose.pose.position.y, 2));                    
                     
+                    // calculating vehicle heading from gps point n and gps point n-1
+                    vehicle_heading_from_gps = std::atan2((geodesy_msg.pose.pose.position.y - previous_geodesy_point.y), (geodesy_msg.pose.pose.position.x - previous_geodesy_point.x));
+
+                    // applying a low-pass filter and creating quaternion message to stuff into the geodesy_tf message
+                    if(second_message_sent)
+                    {
+                        vehicle_heading_from_gps = lpf_coeff * vehicle_heading_from_gps + (1 - lpf_coeff) * previous_vehicle_heading;
+
+                        // heading_deque.push_back(vehicle_heading_from_gps);

```

Comment:
Consider making this it's own function

Reply:
I'm just going to remove the low-pass filter and median filter functionality from the code. We most likely won't use either of the two filters as we can just depend on the EKF.